webapp/views.py

```diff
@@ -767,6 +767,116 @@ def post_advantage_subscriptions(preview):
     return flask.jsonify(purchase), 200
 
 
+def cancel_advantage_subscriptions():
+    api_url = flask.current_app.config["CONTRACTS_LIVE_API_URL"]
+
+    if flask.request.args.get("test_backend", False):
+        api_url = flask.current_app.config["CONTRACTS_TEST_API_URL"]
+
+    user_token = flask.session.get("authentication_token")
+
+    if user_info(flask.session):
+        advantage = AdvantageContracts(
+            session,
+            user_token,
+            token_type=("Macaroon" if user_token else "Bearer"),
+            api_url=api_url,
+        )
+    else:
+        return flask.jsonify({"error": "authentication required"}), 401
+
+    payload = flask.request.json
+
+    if not (
+        payload.get("account_id")
+        and payload.get("previous_purchase_id")
+        and payload.get("products")
+    ):
+        return flask.jsonify({"error": "bad request"}), 400
+
+    account_id = payload.get("account_id")
+    previous_purchase_id = payload.get("previous_purchase_id")
+
+    try:
+        monthly_subscriptions = (
+            advantage.get_account_subscriptions_for_marketplace(
+                account_id=account_id,
+                marketplace="canonical-ua",
+                filters={"status": "active", "period": "monthly"},
+            )
+        )
+    except HTTPError:
+        flask.current_app.extensions["sentry"].captureException(
+            extra={"payload": payload}
+        )
+        return (
+            flask.jsonify(
+                {"error": "could not retrieve account subscriptions"}
+            ),
+            500,
+        )
+
+    if not monthly_subscriptions:
```

Comment:
Shouldn't you check monthly_subscriptions["subscriptions"] ?
What do you think about:
```python
subs = monthly_subscriptions.get("subscriptions") if monthly_subscriptions else None
id not subs:
         return flask.jsonify({"error": "no monthly subscriptions found"}), 400
monthly_subscription = subs[0]
```

Reply:
I changed `if not monthly_subscriptions` to `if not monthly_subscriptions.get("subscriptions")`, well spotted. 

monthly_subscriptions.get("subscriptions") already returns `None` by default if it cannot find the key in the dictionary, so the if-else is not needed.